TeachingRecordSystem/src/TeachingRecordSystem.Api/V3/Implementation/Operations/SetProfessionalStatus.cs

```diff
@@ -0,0 +1,606 @@
+                            return new SetProfessionalStatusResult(); //No-op
+                        case ProfessionalStatusStatus.Deferred:
+                        case ProfessionalStatusStatus.InTraining:
+                        case ProfessionalStatusStatus.UnderAssessment:
+                            return ApiError.UnableToChangeFailProfessionalStatusStatus();
+                    }
+                    break;
+                case dfeta_ITTResult.Withdrawn:
+                    switch (command.Status)
+                    {
+                        case ProfessionalStatusStatus.Withdrawn:
+                            return new SetProfessionalStatusResult(); //No-op
+                        case ProfessionalStatusStatus.Deferred:
+                            return ApiError.UnableToChangeWithdrawnProfessionalStatusStatus();
+                    }
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        itt!.dfeta_EstablishmentId = providerId?.ToEntityReference(Account.EntityLogicalName);
+        itt.dfeta_ProgrammeStartDate = command.TrainingStartDate.ToDateTimeWithDqtBstFix(isLocalTime: true);
+        itt.dfeta_ProgrammeEndDate = command.TrainingEndDate.ToDateTimeWithDqtBstFix(isLocalTime: true);
+        itt.dfeta_ProgrammeType = mappedIttProgrammeType;
+        itt.dfeta_CohortYear = cohortYear;
+        itt.dfeta_Subject1Id = subjectId1?.ToEntityReference(dfeta_ittsubject.EntityLogicalName);
+        itt.dfeta_Subject2Id = subjectId2?.ToEntityReference(dfeta_ittsubject.EntityLogicalName);
+        itt.dfeta_Subject3Id = subjectId3?.ToEntityReference(dfeta_ittsubject.EntityLogicalName);
+        itt.dfeta_AgeRangeFrom = ageRange is not null ? ageRange!.Value.From : null;
+        itt.dfeta_AgeRangeTo = ageRange is not null ? ageRange!.Value.To : null;
+        itt.dfeta_Result = ittResult;
+        itt.dfeta_ITTQualificationId = ittQualificationId?.ToEntityReference(dfeta_ittqualification.EntityLogicalName);
+        itt.dfeta_CountryId = countryId?.ToEntityReference(dfeta_country.EntityLogicalName);
+
+        var qtsRegistration = !isNewItt ? lookupData.QtsRegistrations.FirstOrDefault(qts => qts.dfeta_qtsregistrationId == itt!.dfeta_qtsregistration.Id) : null;
+        if (qtsRegistration is null)
+        {
+            // Programme type could change in this API call so need to match against original one if there is a matching QTS record
+            var compatibleQtsRegistrations = SelectCompatibleQtsRegistrationRecords(
+                lookupData.QtsRegistrations,
+                existingProgrammeType ?? (mappedIttProgrammeType.HasValue ? mappedIttProgrammeType!.Value : null),
+                isEarlyYears,
+                lookupData.EarlyYearsTraineeStatusId,
+                lookupData.AorCandidateTeacherStatusId,
+                lookupData.TraineeTeacherDmsTeacherStatusId);
+
+            if (compatibleQtsRegistrations.Count() > 1)
```

Comment:
Do we not consider the link between QTS and ITT at all (if present)?

Reply:
it's on line 245 .. it only tries to find compatible QTS if there isn't a direct link